lib/super_digit.rb

```diff
@@ -1,15 +1,33 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n log n) where n is the value of the passed in integer (and thus determines if the helper will be called... uh idk why I put "n" as the coefficient I just just know it is a variable that will determine if the helper method is called, potentially multiple times...)
+# Space Complexity - O(n log n)
 def super_digit(n)
-   
+  if n < 10
+    return n
+  else
+    return super_digit_helper(n, 0)
+  end
+end
+
+def super_digit_helper(n, sum)
+  sum += n % 10
+  if n < 10
+    return super_digit(sum)
+  else
+    return super_digit_helper(n / 10, sum)
+  end
 end
   
 
-# Time Complexity - ?
-# Space Complexity - ?
+# Time Complexity - O(n log n) for the same half baked reason as above
+# Space Complexity - O(n log n)
 def refined_super_digit(n, k)
```

Comment:
This works, but it's O( (nk)^2) in time complexity where n is the number of digits in n and k is just the k parameter. 

You can do better with dynamic programming.